includes/content-distribution/class-linked-post.php

```
@@ -0,0 +1,395 @@
+<?php
+/**
+ * Newspack Network Linked Post.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack_Network\Content_Distribution;
+
+use Newspack_Network\Debugger;
+use Newspack_Network\Content_Distribution;
+use WP_Post;
+use WP_Error;
+
+/**
+ * Linked Post Class.
+ */
+class Linked_Post {
+	/**
+	 * Post meta key containing the distributed post ID that is unique accross
+	 * the network.
+	 */
+	const NETWORK_POST_ID_META = 'newspack_network_post_id';
+
+	/**
+	 * Post meta key containing the distributed post full payload.
+	 */
+	const PAYLOAD_META = 'newspack_network_post_payload';
+
+	/**
+	 * Post meta key to determine whether the post is unlinked.
+	 */
+	const UNLINKED_META = 'newspack_network_post_unlinked';
+
+	/**
+	 * Post meta key for linked attachments.
+	 */
+	const ATTACHMENT_META = 'newspack_network_linked_attachment';
+
+	/**
+	 * The post ID.
+	 *
+	 * @var int
+	 */
+	public $ID = 0;
+
+	/**
+	 * The network post ID.
+	 *
+	 * @var string
+	 */
+	public $network_post_id = '';
+
+	/**
+	 * The post object.
+	 *
+	 * @var WP_Post
+	 */
+	private $post = null;
+
+	/**
+	 * The distributed post payload.
+	 *
+	 * @var array
+	 */
+	private $payload = [];
+
+	/**
+	 * Constructor.
+	 *
+	 * @param array $payload The distributed post payload.
+	 *
+	 * @throws \InvalidArgumentException If the payload is invalid or the post is
+	 *                                   not configured for distribution.
+	 */
+	public function __construct( $payload ) {
+		$error = self::get_payload_error( $payload );
+
+		if ( is_wp_error( $error ) ) {
+			throw new \InvalidArgumentException( esc_html( $error->get_error_message() ) );
+		}
+
+		$this->payload         = $payload;
+		$this->network_post_id = $payload['config']['network_post_id'];
+
+		$post = $this->query_linked_post();
+		if ( $post ) {
```

Would it be better to throw an error here or is there a use case for having no post?

---

The `Linked_Post` can - and will - be instantiated without the post existing. It will exist once `->insert()` is called.